### DISCUSSION TITLE
Can't update a file in the bucket

### DISCUSSION THREAD

**Discussion by ricardosikic** [supabase/supabase#14025] [Questions]
Hi, I've been trying to update a single file within a folder in the bucket that i created but when i go to update with a new image from the front does not update the image and gives the following error.
`statusCode: "404", error: "Not found"`
Then when i went to check the route of the call it seems to be wrong `http://localhost:54321/storage/v1/object/xxx-images/xxx_image/36a75358-8612-4e0e-96d8-76e4f579b80d.jpg`. So after checked the route i went to the bucket and copy the route from the image uploaded and the route is different from the route of the call previously mentioned `http://localhost:54321/storage/v1/object/public/xxx-images/xxx_image/05f9d208-bab2-45ab-a355-71ba4a41cf59.jpg`.
As you can see the route from the image saved on the bucket has the `/public/` word on the route. I took the api call to update from the documentation [docs](https://supabase.com/docs/reference/javascript/v1/storage-from-update) and the api call on the front that i have made is the following:

```
     const { data, error } = await supabase
      .storage
      .from('xxx-images')
      .update(`xxx_image/${oldPresentationImage}`, presentationImage.file, {
        contentType: 'image/png'
      });
    if(error) {console.log('error ', error)};
```

I don't get why the route is not the same when i call the api update method from the front.  I'm using the version `1.1.0` and i got the policies for `authenticated users`

**Comment by GaryAustin1** [comment]
The object/public/ part should only be used for a public URL fetch of the file in a public bucket.  All API operations should have object/bucket/pathname.  So your update should not use /public/ and just the bucket then the pathname as you show in your update.

Also you must meet select and update policies to do an update even in a public bucket.

**Reply by GaryAustin1** [reply]
Please reply to answers and don't start new answers unless posting a potential solution.

You have to also have select checked in order for some of them to get highlighted with update selected.  Update policy by itself will not work.

**Reply by ricardosikic** [reply]
yes, sorry about that, i did select `select and update` thanks for that. But the /public/. Sorry I did not understand about not to use /public/ on the api call. When i call the update function the console error shows the following `/storage/v1/object/xxx-images/xxx_image/a0b0b513-cba0-416a-bbf6-5dc8b98c121c.jpg` it has the /object/bucket/pathname

**Reply by GaryAustin1** [reply]
You don't use /public/ at all in storage API calls (unless you have a bucket named public).  You do use public in the url, if the bucket is public and you want to bypass RLS, and just use the url like in an image src string.

**Reply by ricardosikic** [reply]
Hello @GaryAustin1 thanks for the answer, if I update the existing file of the bucket from the front the new one will have the new file name on the bucket or it will keep it?
